### PR TITLE
Pressreader Editions | Native App / OS Detection

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

Unlike other apps, within Okta, the Pressreader Editions app for both Android and iOS is set up as a single application, this was done to mirror the already existing Editions app.

However we need to know whether the user is using the iOS or Android version of the app (`isNativeApp` variable) to show app specific design variants and hide the CMP banner, as well as displaying the name of the app if the user is signed in already (`appName` variable).

Thankfully we already use [`Bowser`](https://github.com/bowser-js/bowser) to parse the `user-agent` header we get from the client.

So we can use a combination of the `appLabel` (`editions_pressreader`) and the user agent to determine whether we're on iOS or Android, and derive the correct app name!

## Tested

- [x] CODE

### CODE REVIEW

**Review by akinsola-guardian — DISMISSED** [review]
LGTM

File: src/shared/lib/appNameUtils.ts
Comment (pvighi) [inline src/shared/lib/appNameUtils.ts:82]: is this strictly true ? wouldn't this just return whether you are using a mobile device? even if you don't have the app installed this would return true if you use your phone browser, right ? Do we not care about this distinction ?

Comment (coldlink) [reply src/shared/lib/appNameUtils.ts]: Have updated both the name and comment to make it clearer.

File: src/shared/lib/appNameUtils.ts
Comment (pvighi) [inline src/shared/lib/appNameUtils.ts:86]: I still think this is slightly confusing because it mentions Native when the code doesn't have anything to do with native app. Is Native os a term people use ? because the way I understand it it's  just the client apps that can be native or running on a browser

But I guess the return type is `IsNativeApp`.  Also it's pretty strange to me that a type with that name is not a boolean but that is not introduced in this PR.

In any case this is all about naming so I'll approve and leave it up to you if you are happy to merge it like this

**Review by pvighi — APPROVED** [review]

**Pull request merged**
